arch/cortex-m/src/syscall.rs

```diff
@@ -0,0 +1,183 @@
+//! Implementation of the architecture-specific portions of the kernel-userland
+//! system call interface.
+
+use core::ptr::{read_volatile, write_volatile};
+
+use kernel;
+
+/// This is used in the syscall handler. When set to 1 this means the
+/// svc_handler was called.
+// #[allow(private_no_mangle_statics)]
+#[no_mangle]
+#[used]
+pub static mut SYSCALL_FIRED: usize = 0;
```

Comment:
does this need to be public?

Reply:
It's public because it's used in the cortex-m4 crate but defined in the cortex-m crate. I'm not sure if it is strictly required.